scripts/pylib/twister/twisterlib/coverage.py

```diff
@@ -278,6 +283,9 @@ def add_ignore_file(self, pattern):
     def add_ignore_directory(self, pattern):
         self.ignores.append(".*/" + pattern + '/.*')
 
+    def add_ignore_branch_pattern(self, pattern):
+        self.ignore_branch_patterns.append(pattern)
+
```

Comment:
Couldn't this be moved to `CoverageTool` together with `self.ignore_branch_patterns` to make use of inheritance (both `Gcovr` and `Lcov` inherit from it) and reduce code duplication?

Reply:
It probably could, I just copied the pattern from the other ignore options

Reply:
Other ignore options actually differ between `Lcov` and `Gcovr` (`*` vs `.*`)